pkg/apputils/utils.go

```diff
@@ -174,7 +174,12 @@ func getGitClients(ctx context.Context, url, configUrl, namespace string, isHelm
 }
 
 func getAuthService(ctx context.Context, providerUrl string) (auth.AuthService, error) {
-	gitProvider, providerErr := auth.GetGitProvider(ctx, providerUrl)
+	normalizedUrl, err := gitproviders.NewNormalizedRepoURL(providerUrl)
+	if err != nil {
+		return nil, fmt.Errorf("error normalizing url: %w", err)
+	}
+
+	gitProvider, providerErr := auth.GetGitProvider(ctx, normalizedUrl.String())
```

Comment:
Might be an opportunity to change the `GetGitProvider` to accept a `NormalizedRepoURL`. It looks like that function is doing more work than it has to:

https://github.com/weaveworks/weave-gitops/blob/ae571fbc056cb5cd341a9f99b029b7f6f92c6258/pkg/services/auth/auth.go#L49